src/wrapper.rs

```diff
@@ -608,23 +619,33 @@ pub struct Dxil {
 }
 
 impl Dxil {
-    pub fn new() -> Result<Self, HassleError> {
-        #[cfg(not(windows))]
-        {
-            Err(HassleError::WindowsOnly(
-                "DXIL Signing is only supported on windows at the moment".to_string(),
-            ))
-        }
+    #[cfg(not(windows))]
+    pub fn new(_: Option<&Path>) -> Result<Self, HassleError> {
+        Err(HassleError::WindowsOnly(
+            "DXIL Signing is only supported on Windows".to_string(),
+        ))
+    }
+
+    #[cfg(windows)]
+    pub fn new(lib_path: Option<&Path>) -> Result<Self, HassleError> {
+        let joined_path;
```

Comment:
We probably shouldn't have this `joined_path` hack but instead just return owned objects from the if-statements.

Reply:
It's a bit of a hack, but also disappointing that we can't artificially enlarge the lifetime of `joined_path` without unnecessarily cloning the others.

If we do it like this I'll probably change the signatures to take `Option<PathBuf>` so that the caller can at least move an owned path if they have one instead of always cloning.